corehq/apps/locations/ucr_expressions.py

```diff
@@ -85,14 +86,9 @@ def configure(self, location_id_expression, location_type_expression):
     def __call__(self, item, context=None):
         location_id = self._location_id_expression(item, context)
         location_type = self._location_type_expression(item, context)
+        return self._get_ancestor(location_id, location_type)
 
-        cache_key = (self.__class__.__name__, location_id, location_type)
-        if context.get_cache_value(cache_key, False) is not False:
-            return context.get_cache_value(cache_key)
-        ancestor_json = self._get_ancestor(location_id, location_type)
-        context.set_cache_value(cache_key, ancestor_json)
-        return ancestor_json
-
+    @ucr_context_cache(vary_on=('location_id', 'location_type',))
     def _get_ancestor(self, location_id, location_type):
         try:
             location = SQLLocation.objects.get(location_id=location_id)
```

Comment:
Perhaps worth caching both of these calls separately (getting the location and the ancestory), out of scope for this PR

Reply:
Agreed.  The data source I was looking at makes a bunch of DB hits which I think come through here - I'll be looking at this next.